server/src/HttpServer.hs

```diff
@@ -25,6 +26,22 @@ import qualified Store
 import qualified Core
 import qualified Metrics
 
+import qualified Prometheus
+
+newtype Action e m a = Action { runAction :: Scotty.ActionT e m a }
+
+instance (Scotty.ScottyError e, MonadIO m) => Functor (Action e m) where
+  fmap f = Action . fmap f . runAction
+instance (Scotty.ScottyError e, MonadIO m) => Applicative (Action e m) where
+  pure = Action . return
+  liftA2 f a b = Action $ liftA2 f (runAction a) (runAction b)
+instance (Scotty.ScottyError e, MonadIO m) => Monad (Action e m) where
+  m >>= f = Action $ runAction m >>= runAction . f
+instance (Scotty.ScottyError e, MonadIO m) => MonadIO (Action e m) where
+  liftIO = Action . Scotty.liftAndCatchIO
+instance (Scotty.ScottyError e, MonadIO m) => Prometheus.MonadMonitor (Action e m) where
+  doIO = liftIO
```

Comment:
What's this needed for?

Reply:
In [observeDuration](https://hackage.haskell.org/package/prometheus-client-1.1.0/docs/Prometheus.html#v:observeDuration), we need the monad to be an instance of `MonadMonitor`, that's why I wrapped it. If I take the middleware way, I *may* not have to do this though. I'll check it out.